### PR TITLE
Crystal action space with example problem

### PR DESCRIPTION
**Pull request by jlaw9**
For now, just works with MCTS and a dummy reward function.

### CODE REVIEW

File: examples/crystal_volume/optimize_crystal_volyme.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    G2 = nx.read_edgelist(action_graph2_file, delimiter='\t', data=False, create_using=G2)
+    print(f'{G2.number_of_nodes()} nodes, {G2.number_of_edges()} edges')
+
+    # also load the mapping from composition to composition type
+    working_dir = 'inputs'
+    df_comp = pd.read_csv(f'{working_dir}/compositions.csv.gz')
+    print(df_comp.head())
+    compositions = df_comp['composition'].to_list()
+    comp_types = set(df_comp['comp_type'].to_list())
+    comp_to_comp_type = dict(zip(df_comp['composition'], df_comp['comp_type']))
```
Comment (pstjohn) [inline examples/crystal_volume/optimize_crystal_volyme.py:79]: Does this need to get spelled out in a csv? I would have thought that getting the composition type (i.e., _1_1_4) from a composition (i.e., Li1Sc1F4) would be pretty straightforward if the composition strings are reasonable

File: examples/crystal_volume/crystal_state.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    ) -> None:
+        """
+        :param builder: A CrystalBuilder class
+        :param action_node: A representation of the current state in one of the action graphs
+        :param terminal: Whether this state is a decoration of a specific structure (i.e., final state)
+        """
+        self._builder: any = builder
+        self._elements: str = elements
+        self._composition: str = composition
+        self._comp_type: str = comp_type
```
Comment (pstjohn) [inline examples/crystal_volume/crystal_state.py:33]: If we had the composition we should be able to fill these other two out, right?

File: examples/crystal_volume/optimize_crystal_volyme.py (hunk 1 above)
Comment (jlaw9) [reply examples/crystal_volume/optimize_crystal_volyme.py:79]: Yes that is true. If we assume the composition convention won't change, then it's not that hard.

File: examples/crystal_volume/crystal_state.py (hunk 2 above)
Comment (jlaw9) [reply examples/crystal_volume/crystal_state.py:33]: Yes. At the beginning there are element combinations that are not yet compositions, but I can handle those cases easily enough

**Comment by pstjohn** [comment]
we'll want to rebase this on devel after the squash and merge

**Pull request merged**